src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/Runtime/PreferAsSpanOverSubstring.cs

```diff
@@ -0,0 +1,312 @@
+                IMethodSymbol asSpan1, IMethodSymbol asSpan2)
+            {
+                StringType = stringType;
+                RoscharType = roscharType;
+                MemoryExtensionsType = memoryExtensionsType;
+                Substring1 = substring1;
+                Substring2 = substring2;
+                AsSpan1 = asSpan1;
+                AsSpan2 = asSpan2;
+            }
+
+            public static bool TryGetSymbols(Compilation compilation, out RequiredSymbols symbols)
+            {
+                var stringType = compilation.GetSpecialType(SpecialType.System_String);
+                var charType = compilation.GetSpecialType(SpecialType.System_Char);
+                var int32Type = compilation.GetSpecialType(SpecialType.System_Int32);
+
+                if (stringType is null || charType is null || int32Type is null)
+                {
+                    symbols = default;
+                    return false;
+                }
+
+                var roscharType = compilation.GetOrCreateTypeByMetadataName(WellKnownTypeNames.SystemReadOnlySpan1)?.Construct(charType);
+                var memoryExtensionsType = compilation.GetOrCreateTypeByMetadataName(WellKnownTypeNames.SystemMemoryExtensions);
+
+                if (roscharType is null || memoryExtensionsType is null)
+                {
+                    symbols = default;
+                    return false;
+                }
+
+                var int32ParamInfo = ParameterInfo.GetParameterInfo(int32Type);
+                var stringParamInfo = ParameterInfo.GetParameterInfo(stringType);
+
+                var substringMembers = stringType.GetMembers(nameof(string.Substring)).OfType<IMethodSymbol>();
+                var substring1 = substringMembers.GetFirstOrDefaultMemberWithParameterInfos(int32ParamInfo);
+                var substring2 = substringMembers.GetFirstOrDefaultMemberWithParameterInfos(int32ParamInfo, int32ParamInfo);
+
+                var asSpanMembers = memoryExtensionsType.GetMembers(nameof(MemoryExtensions.AsSpan)).OfType<IMethodSymbol>();
+                var asSpan1 = asSpanMembers.GetFirstOrDefaultMemberWithParameterInfos(stringParamInfo, int32ParamInfo);
+                var asSpan2 = asSpanMembers.GetFirstOrDefaultMemberWithParameterInfos(stringParamInfo, int32ParamInfo, int32ParamInfo);
+
+                if (substring1 is null || substring2 is null || asSpan1 is null || asSpan2 is null)
+                {
+                    symbols = default;
+                    return false;
+                }
+
+                symbols = new RequiredSymbols(
+                    stringType, roscharType,
+                    memoryExtensionsType,
+                    substring1, substring2,
+                    asSpan1, asSpan2);
+                return true;
+            }
+
+            public INamedTypeSymbol StringType { get; }
+            public INamedTypeSymbol RoscharType { get; }
```

Comment:
[minor] Could we use a more explicit name? I had to look at the creation of the `roscharType` variable to understand what this is about.

Reply:
Fixed. I'll also fix this on my other PRs.